### PR TITLE
Upgrades Cloudflare addon.

### PR DESCRIPTION
**Pull request by hellofromtonya**
This PR upgrades to use our [Cloudflare package](https://github.com/wp-media/cloudflare).

- Removes the old Cloudflare code and tests.
- Adds the new Cloudflare package into `inc/Addons`.
- Wires it into the container.

## TODO

- [x] Add integration tests to ensure Cloudflare works as part of the integrated Rocket Addon.

### CODE REVIEW

File: tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		parent::setUpBeforeClass();
+
+		self::$api_credentials_config_file = WP_ROCKET_PLUGIN_ROOT . '/tests/env/local/cloudflare.php';
+
+		self::$container = apply_filters( 'rocket_container', '' );
+		self::setApiCredentials();
+		self::resetTransients();
+		self::$subscriber = self::$container->get( 'cloudflare_subscriber' );
+
+		// Store original state. Why? We'll restore it before existing this class.
```
Comment (Tabrisrp) [inline tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php]:
```suggestion
		// Store original state. Why? We'll restore it before exiting this class.
```

File: tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		$class    = new ReflectionClass( self::$subscriber );
+		self::$cf_property = $class->getProperty( 'cloudflare' );
+		self::$cf_property->setAccessible( true );
+		self::$subscriber_cf = self::$cf_property->getValue( self::$subscriber );
+		self::$options_property            = $class->getProperty( 'options' );
```
Comment (Tabrisrp) [inline tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php]:
```suggestion
		self::$options_property = $class->getProperty( 'options' );
```

Comment (hellofromtonya) [reply tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php]: Applied this after I renamed the folder. https://github.com/wp-media/wp-rocket/pull/2313/commits/027d1c830137452b046dc34c796f8a5a94ed8c76

File: tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php (hunk 1 above)
Comment (hellofromtonya) [reply tests/Integration/inc/Addons/Cloudflare/CloudflareSubscriber/TestCase.php]: Applied this after I renamed the directory. https://github.com/wp-media/wp-rocket/pull/2313/commits/027d1c830137452b046dc34c796f8a5a94ed8c76

**Pull request merged**